src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -129,6 +142,28 @@ public Future<?> handle(String deploymentId, String deploymentApi) {
                 });
     }
 
+    private Future<?> handleInterceptor(String deploymentApi) {
+        ApiKeyData apiKeyData = context.getApiKeyData();
+        List<String> interceptors = context.getInterceptors();
+        int nextIndex = apiKeyData.getInterceptorIndex() + 1;
+        if (nextIndex < interceptors.size()) {
+            String interceptorName = interceptors.get(nextIndex);
+            Interceptor interceptor = context.getConfig().getInterceptors().get(interceptorName);
+            if (interceptor == null) {
+                log.warn("Interceptor is not found for the given name: {}", interceptorName);
+                return respond(HttpStatus.NOT_FOUND, "Interceptor is not found");
+            }
+            context.setDeployment(interceptor);
+
+            setupProxyApiKeyData();
+
+            InterceptorController controller = new InterceptorController(proxy, context);
+            return controller.handle();
+        } else { // all interceptors are completed we should call the initial deployment
+            return handleDeployment(apiKeyData.getInitialDeployment(), deploymentApi);
```

Comment:
Should we take deploymentApi from initial request as well?
It's possible that interceptor call dial-core with invalid or wrong deployment API

Reply:
No it will never happen. The interceptor should not change the original deployment API.

Reply:
fixed